pkg/scheduler/plugins/resource-strategy-fit/resource_strategy_fit.go

```diff
@@ -104,6 +105,19 @@ func calculateWeight(args framework.Arguments) ResourceStrategyFit {
 	}
 
 	for k, v := range resources {
+		configKeyStr := string(k)
+
+		if strings.Contains(configKeyStr, "*") {
```

Comment:
Why do we need to add a `strings.Contains` check in the outer layer? Shouldn't this be put in `isValidWildcardPattern`?